ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Mapping
+
+from ndcube import NDCube
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, NDCube):
+        super().__init__()
+
+        # Set values using the setters for validation
+        self.mapping = {}
```

Comment:
The internal state modified by add and remove should be hidden, and needs to be separate from the total computed state.
```suggestion
        self._internal_coords = {}
        
    @property
    def _all_coords(self):
        return self._internal_coords
```

If you then use `_all_coords` everywhere you are currently using `.mapping` (with the excpetion of add and remove which should mutate `_internal_coords`) we can modify the property here to extract coords from the wcs and extra coords in followup PRs.